src/Analyzers/CSharp/Analyzers/UseExpressionBodyForLambda/UseExpressionBodyForLambdaDiagnosticAnalyzer.cs

```diff
@@ -31,33 +31,17 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(AnalyzeIfEnabled,
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax,
                 SyntaxKind.SimpleLambdaExpression, SyntaxKind.ParenthesizedLambdaExpression);
 
-        private static void AnalyzeIfEnabled(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
         {
             var analyzerOptions = context.Options;
             var syntaxTree = context.SemanticModel.SyntaxTree;
             var optionValue = UseExpressionBodyForLambdaHelpers.GetCodeStyleOption(analyzerOptions.GetAnalyzerOptions(syntaxTree));
-            var severity = UseExpressionBodyForLambdaHelpers.GetOptionSeverity(optionValue);
```

Comment:
What happens for `ReportDiagnostic.Suppress`? We want to make sure we don't offer the code fix, but instead offer a refactoring that goes in the opposite direction.

Reply:
I think we won't produce a codefix as we report the diagnostic with IsSuppressed to the compiler https://github.com/dotnet/roslyn/blob/1c3cdb0b85a8496a8f5108faab046924360b6017/src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs#L248

We're investigating a different approach than IsSuppressed in another PR, but we'll make sure that the new approach preserves this (current direction is to not report a diagnostic at all).

Reply:
ReportDiagnostic.Hidden is questionable.

Reply:
Changed Hidden behavior for the refactoring in 9997800. Hope it's the correct decision.